### PR TITLE
 refactor(egen): follows new template+sentence and other minor corrections

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
This PR includes the following minor updates:

1. Adds Colab Enterprise link.
2. Corrects some sentences.
3. Corrects the structure according to the template.
4. Contracts words wherever needed.
5. Replaces REGION with LOCATION.
6. Removes unnecessary sections.
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

**Comment by katiemn** [comment]
Is this draft PR ready for review?

**Comment by katiemn** [comment]
If so, can you please refactor the title to the following format: "refactor(egen): _your PR comment_"

**Comment by krishr2d2** [comment]
> Is this draft PR ready for review?

Not yet. 
The notebook involves some IS_TESTING conditions outside the cleaning up section. Waiting on an advise to keep or remove them.
image

tagging @gericdong

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 1]
```diff
@@ -105,305 +108,252 @@
...
       "metadata": {
-        "id": "4f4512bf63b3"
+        "id": "dd28c9e4f067"
       },
       "source": [
-        "#### Region\n",
+        "### Set Google Cloud project information\n",
         "\n",
-        "You can also change the `REGION` variable used by Vertex AI. Learn more about [Vertex AI regions](https://cloud.google.com/vertex-ai/docs/general/locations)."
+        "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:225]: change to:
- "Learn more about how to [set up a project and a development environment]..."

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 2]
```diff
@@ -105,305 +108,252 @@
...
-      "metadata": {
-        "id": "NIq7R4HZCfIc"
+        "id": "lR6Wwv-hCCN-"
       },
-      "outputs": [],
       "source": [
-        "! gsutil mb -l {REGION} -p {PROJECT_ID} {BUCKET_URI}"
+        "## Prepare the data\n",
+        "\n",
+        "For this tutorial, you use the [Stack Overflow dataset](https://console.cloud.google.com/marketplace/product/stack-exchange/stack-overflow) of question and answers hosted on BigQuery.\n",
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:345]: Don't need "you". Note that "use" is a directive and implies "you," the reader of the document.
- "For this tutorial, use the..."

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 3]
```diff
@@ -105,305 +108,252 @@
...
+        "\n",
+        "> This public dataset is hosted in Google BigQuery and is included in BigQuery's 1TB/mo of free tier processing. This means that each user receives 1TB of free BigQuery processing every month, which can be used to run queries on this public dataset."
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:347]: Don't need "This means that":

- "Each user receives 1TB of free BigQuery processing every month, which can be used to run queries on this public dataset"

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 4]
```diff
@@ -470,13 +418,13 @@
...
         "\n",
         "Use the [sentence-t5 encoder](https://tfhub.dev/google/sentence-t5/st5-base/1) developed by Google for converting text to embeddings.\n",
         "\n",
         "> The sentence-T5 family of models encode text into high-dimensional vectors that can be used for text classification, semantic similarity, clustering and other natural language processing tasks.\n",
         ">\n",
-        "> Our model is built on top of T5 (i.e. the Text-To-Text Transfer Transformer). It is trained on a variety of data sources and initialized from pre-trained T5 models with different model sizes. The input is variable-length English text and the output is a 768-dimensional vector. The sentence-T5 base model employs a 12-layer transformer architecture as the T5 base model does."
+        "> Our model is built on top of T5 (i.e. the Text-To-Text Transfer Transformer). It's trained on a variety of data sources and initialized from pre-trained T5 models with different model sizes. The input is variable-length English text and the output is a 768-dimensional vector. The sentence-T5 base model employs a 12-layer transformer architecture as the T5 base model does."
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:427]: Don't use "Our". Suggest changing to "The model". Also, should be i.e.,:
- "The model is built on top of T5 (i.e., the Text-To-Text Transfer Transformer). "

Redo do the last sentence:
- "The sentence-T5 base model employs a 12-layer transformer architecture as does the T5 base model."

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 5]
```diff
@@ -819,7 +742,7 @@
       "source": [
         "### Create VPC peering\n",
         "\n",
-        "For simplicity, we setup VPC peering to the ucaip-haystack-vpc-network network. You can create a different network for your project.\n",
+        "For simplicity, you setup VPC peering to the `ucaip-haystack-vpc-network` network. You can create a different network for your project.\n",
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:745]: again, don't need "you" here. Also "setup" is a noun. Should be "set up", which is a verb.
- "For simplicity, set up VPC peering to the `ucaip-haystack-vpc-network` network."

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 6]
```diff
@@ -856,7 +779,7 @@
         "\n",
         "Next, create the connection for VPC peering.\n",
         "\n",
-        "*Note:* If you get a PERMISSION DENIED, you may not have the neccessary role 'Compute Network Admin' set for your default service account. In the Cloud Console, do the following steps.\n",
+        "**Note:** If you get a PERMISSION DENIED, you may not have the neccessary 'Compute Network Admin' role set for your default service account. In the Cloud Console, do the following steps.\n",
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:782]: Don't need "steps" here.

**Review by kittyabs — COMMENTED** [review]
some suggested edits

**Comment by krishr2d2** [comment]
The redis instance creation step fails as the service account needs additional permissions inside the PR test environment. So, added the IS_TESTING conditions back for the redis steps.

**Review by katiemn — APPROVED** [review]

**Pull request merged**